mozci/task.py

```diff
@@ -69,48 +69,54 @@ def get_artifact(self, path):
 
 @dataclass
 class TestResult:
+    """Contains information relating to a single test failure within a TestTask."""
     group: str
     test: str
     ok: bool
 
 
 @dataclass
 class TestTask(Task):
+    """Subclass containing additional information only relevant to 'test' tasks."""
     _results: List[TestResult] = field(default_factory=list)
     _groups: List = field(default_factory=list)
 
-    def _load_errorsummary(self):
+    @memoized_property
```

Comment:
I didn't do this in my PR as I wanted to avoid using too much memory when you have thousands of tasks loaded

Reply:
Good point.

We actually also `memoize` the `get_artifact` function. So no need to memoize this value again.. the cost of splitting it into lines isn't terrible.